apps/muddle-probe/main.cpp

```diff
@@ -0,0 +1,319 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "crypto/ecdsa.hpp"
+#include "core/serializers/main_serializer.hpp"
+#include "core/byte_array/byte_array.hpp"
+#include "core/byte_array/decoders.hpp"
+#include "core/service_ids.hpp"
+#include "core/commandline/parameter_parser.hpp"
+#include "network/fetch_asio.hpp"
+#include "network/tcp/client_implementation.hpp"
+#include "muddle/packet.hpp"
+#include "muddle/network_id.hpp"
+
+#include <iostream>
+#include <string>
+
+struct PingMessage {};
+
+namespace fetch {
+namespace serializers {
+
+template <typename D>
+struct MapSerializer<PingMessage, D>
+{
+public:
+  using Type       = PingMessage;
+  using DriverType = D;
+  using EnumType   = uint64_t;
+
+  static const uint8_t TYPE = 1;
+
+  template <typename T>
+  static void Serialize(T &map_constructor, Type const &msg)
+  {
+    FETCH_UNUSED(msg);
+
+    auto map = map_constructor(1);
+    map.Append(TYPE, uint64_t{0});
+  }
+
+  template <typename T>
+  static void Deserialize(T &map, Type &msg)
+  {
+    FETCH_UNUSED(map);
```

Comment:
This seems odd - should this go in?

Reply:
This is to approximate the message generated in muddle. This is a non-critical tool so I think it is fine for the time being